### PR TITLE
Add devbox config

### PR DESCRIPTION
**Pull request by hardyoyo**
Adds configuration and changes a few things to facilitate using [Jetify Devbox](https://www.jetify.com/devbox) as a dev environment for JSchol. Also contains some changes to SCSS files to handle some deprecation warnings that we see during deploys and while running the dev environment.

### CODE REVIEW

File: setup.sh [hunk 1]
```diff
@@ -1,22 +1,25 @@
...
 printf "== Installing local Ruby gems ==\n"
+gem install bundler
 bundle install --quiet --path=gems --binstubs
 
 #printf "\n== Uninstalling Ruby gems no longer used ==\n"
 bundle clean
 
 printf "\n== Installing node packages (used by gulp and iso via Node) ==\n"
-npm install npm@10.8.1
+npm install npm@10.2.4
```
Comment (everreau) [inline setup.sh:18]: Why is this going back to an older version of npm?

Comment (hardyoyo) [reply setup.sh:18]: Good catch, it should not, I will fix it

**Review by everreau — APPROVED** [review]
👍

File: app/scss/_utilities.scss [hunk 2]
```diff
@@ -225,7 +225,7 @@ Background color with progressively-enhanced blur effect. Example:
 // Fluid text mixin courtesy of: https://css-tricks.com/snippets/css/fluid-typography/
 
 @function strip-unit($value) {
-  @return $value / ($value * 0 + 1);
+  @return calc(#{$value} / (#{$value} * 0 + 1));
```
Comment (myucekul) [inline app/scss/_utilities.scss:228]: Please add comment on why calc is needed. Where is it defined?

File: setup.sh [hunk 3]
```diff
@@ -14,9 +19,7 @@ npm install npm@10.8.1
 npm install
 npm install gulp-cli # shouldn't be necessary, but seems to be
 
-if [[ `/bin/hostname` == "*pub-submit*" ]]; then
```
Comment (myucekul) [inline setup.sh:17]: How do I know this submit specific code is no longer needed? setup.sh is not specific to dev box.

File: .envrc.example [hunk 4]
```diff
@@ -0,0 +1,87 @@
+# .envrc congifuration for jschol
```
Comment (myucekul) [inline .envrc.example:1]: This file looks like config/env.template except filled in. Where is it used?

File: app/scss/_utilities.scss (hunk 2 above)
Comment (hardyoyo) [reply app/scss/_utilities.scss:228]: This was one of the deprecation warnings. The previous version of the code uses division outside of a calc function, and will not be supported in a future version of the Sass library that we use for JSchol. [Calc is a built-in function in Sass](https://sass-lang.com/documentation/values/calculations/).

File: .envrc.example (hunk 4 above)
Comment (hardyoyo) [reply .envrc.example:1]: this is an example direnv configuration, used with direnv

File: setup.sh (hunk 3 above)
Comment (hardyoyo) [reply setup.sh:17]: this is not how the splash page generator is built any more. See https://github.com/eScholarship/cdl-splashgen

File: app/scss/_utilities.scss (hunk 2 above)
Comment (hardyoyo) [reply app/scss/_utilities.scss:228]: here's a [message about the deprecation warnings](https://stackoverflow.com/questions/71262427/deprecation-warning-using-for-division-outside-of-calc-is-deprecated-and-wi), we had a wall of them every time we would build or deploy JSchol.

**Pull request merged**